src/js/components/FilterTree.tsx

```diff
@@ -16,22 +15,80 @@ import Search from "../state/Search"
 import usePopupMenu from "./hooks/usePopupMenu"
 import {remote} from "electron"
 import Last from "../state/Last"
+import {SearchRecord} from "../types"
+
+const getPins = (node?: InvestigationNode): string[] => {
+  const result = reduce(
+    node?.getPath(),
+    (res, n) => {
+      res.push(n.model.filter)
+      return res
+    },
+    []
+  )
+
+  // don't include root, or final path element as pin
+  return tail(initial(result))
+}
+
+const nodeIsActivePin = (node: InvestigationNode, prevPins: string[]) => {
+  const selected = [...getPins(node), node.model.filter]
+  return isEqual(selected, take(prevPins, selected.length))
+}
+
+const nodeIsActive = (
+  node: InvestigationNode,
+  prevPins: string[],
+  prevProgram: string
+) => {
+  return (
+    node &&
+    node.model.filter === prevProgram &&
+    isEqual(prevPins, getPins(node))
+  )
+}
 
-type Props = {node: any; i: number; connId: string; spaceId: string}
+const reconstructSearch = (node: InvestigationNode): SearchRecord => {
+  return {
+    ...node.model.finding.search,
+    program: node.model.filter,
+    pins: getPins(node)
+  }
+}
+
+type Props = {
+  node: InvestigationNode
+  i: number
+  connId: string
+  spaceId: string
+}
 
 function NodeRow({node, i, connId, spaceId}: Props) {
   const dispatch = useDispatch()
   const last = useSelector(Last.getSearch)
-  const pinnedFilters = last?.pins || []
-  const previous = last?.program || ""
+  const prevPins = last?.pins || []
+  const prevProgram = last?.program || ""
   const menu = usePopupMenu([
     {
       label: "Delete",
       click: () => {
-        const multiTs = node.mapChildren((node) => node.data.finding.ts)
-        globalDispatch(
-          Investigation.deleteFindingByTs(connId, spaceId, multiTs)
-        )
+        remote.dialog
+          .showMessageBox({
+            type: "warning",
+            title: "Delete History Entry",
+            message: `Are you sure you want to remove this entry and it's underlying query?`,
+            buttons: ["OK", "Cancel"]
+          })
+          .then(({response}) => {
+            if (response === 0) {
+              const multiTs = node
+                .all(() => true)
```

Comment:
Is this the same as saying `node.children`?

Reply:
As mentioned offline: no, `all()` is similar to a `descendants()` or function which recursively returns all children, grand-children, etc. It just takes a predicate function which I'm not really utilizing any sort of filter here, hence `() => true` (return all nodes)